onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -191,6 +191,8 @@ common::Status NnapiExecutionProvider::Compile(const std::vector<onnxruntime::No
       nnapi::ModelBuilder builder(graph_viewer);
       builder.SetUseNCHW(false);
       builder.SetUseFp16(false);
+      builder.SetTargetDeviceOption(nnapi::ModelBuilder::TargetDeviceOption::ALL_DEVICES);
+
```

Comment:
For future reference I believe the guideline is to keep PRs to one specific topic, so would have been good to separate the changes to ComputePadAndOutputShape which affect other providers from these somewhat orthogonal updates to NNAPI.

Reply:
Yes, you are right, anyway I do not really like the way how devices support is logged, removed this part

---
In reply to: [454724961](https://github.com/microsoft/onnxruntime/pull/4499#discussion_r454724961) [](ancestors = 454724961)